app/model/sdl/Abstract/Model.js

```diff
@@ -1033,6 +1032,54 @@ SDL.SDLModel = Em.Object.extend({
    */
   imageCheckList: {},
 
+  /**
+   * @description Checks images in request provided inside 'objectsWithImages' param.
+   * @param requestID {Integer} Id of request
+   * @param callback {Function} User defined callback 
+   * @param objectsWithImages {Array} Array of data structures, specific to request,
+   *                                  that contain images.
+   * @return {Boolean} Returns true, if all images have valid extension. 
+   *                   Returns false, if at least 1 image has invalid extension.
+   */
+  checkImagesInRequest: function(requestID, callback, objectsWithImages) {
+    if (!Array.isArray(objectsWithImages) || 
+        (Array.isArray(objectsWithImages) && objectsWithImages.length == 0)) {
+      return false;
+    }
+
+    var ImageInvalid = {};
+    var imageList = [];
+
+    var checkExtension = function(img) {
+      if(!img) {
+        return;
+      }
+      if(img.isTemplate && !SDL.NavigationController.isPng(img.value)) {
+        throw ImageInvalid;
+      }
+      imageList.push(img.value);
+    }
+
+    try {
```

Comment:
@YarikMamykin try to avoid try-catch blocks if possible. In your case you can use boolean flag like `var AllImagesAreValid = true` and set it to false in `forEach` block. Then just return false if this flag was set in `forEach` block.

Reply:
@AKalinich-Luxoft take a look at https://github.com/LuxoftSDL/sdl_hmi/pull/9/commits/d9b7d04de4e69380beb0003e6d8aa4e6327e5910